mlc_llm/quantization/ft_quantization.py

```diff
@@ -46,16 +48,22 @@ def f_quantize(bb: relax.BlockBuilder, inputs: List[relax.Expr]):
                 inputs[0],
                 primfunc_name_hint="encode",
             )
+
             packed_weight = bb.normalize(encoded_data[0])
-            encoded_weight = bb.emit(
-                relax.call_pure_packed(
-                    "cutlass.ft_preprocess_weight",
-                    packed_weight,
-                    self.sm,
-                    self.nbit == 4,
-                    sinfo_args=packed_weight.struct_info,
+
+            if self.do_preprocess:
+                encoded_weight = bb.emit(
+                    relax.call_pure_packed(
+                        "cutlass.ft_preprocess_weight",
+                        packed_weight,
+                        self.sm,
+                        self.nbit == 4,
+                        sinfo_args=packed_weight.struct_info,
+                    )
                 )
-            )
+            else:
+                encoded_weight = packed_weight
```

Comment:
We need to skip weight preprocessing during quantization, for FT quantization + `num_shards > 1` case.